lib/engine/game/g_18_usa/game.rb

```diff
@@ -630,20 +630,78 @@ def operating_round(round_num)
             G1817::Step::CashCrisis,
             G18USA::Step::ObsoleteTrain,
             G18USA::Step::Loan,
+            G18USA::Step::ScrapTrain,
             G18USA::Step::SpecialTrack,
             G18USA::Step::SpecialToken,
             G18USA::Step::SpecialBuyTrain,
             G18USA::Step::Assign,
             G18USA::Step::Track,
             G18USA::Step::DenverTrack,
             G18USA::Step::Token,
+            G18USA::Step::BuyPullman,
             G18USA::Step::Route,
             G18USA::Step::Dividend,
-            Engine::Step::DiscardTrain,
+            G18USA::Step::DiscardTrain,
             G18USA::Step::BuyTrain,
           ], round_num: round_num)
         end
 
+        def pullman_scrap_value
+          50
+        end
+
+        def scrap_info
+          "Scrap Pullman for #{format_currency(pullman_scrap_value)}"
+        end
+
+        def scrap_button_text
+          'Scrap Pullman'
+        end
+
+        def use_compact_scrap_trains_view
+          true
+        end
+
+        def use_1840_style_merger_round_scrap_trains?
+          false
+        end
+
+        def crowded_corps
+          @crowded_corps ||= corporations.select do |c|
+            trains = self.class::OBSOLETE_TRAINS_COUNT_FOR_LIMIT ? c.trains.size : c.trains.count { |t| !t.obsolete }
+            trains > train_limit(c) || c.trains.count { |t| pullman_train?(t) } > 1
+          end
+        end
```

Comment:
Is there a reason this crowded_corps logic can't be common? It looks generic. The last line may be something like super || c.trains.count { |t| pullman_train?(t) } > 1.

Reply:
I replaced it with
```

        def crowded_corps
          @crowded_corps ||= super & corporations.select { |c| c.trains.count { |t| pullman_train?(t) } > 1 }
        end

```